Linking issue in xcode

Hey,
it seems there is a problem in building native dependencies on iOS on newest version:

```
Undefined symbols for architecture x86_64:
  "SkFontMgr_New_Custom_Data(sk_sp<SkData>*, int)", referenced from:
      RNSkia::JsiSkFontMgrFactory::FromData(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long) in libreact-native-skia.a(RNSkManager.o)
``` 

Xcode: 13.1
react-native: 0.66.4
react-native-skia 0.1.69

--- a/docs/docs/getting-started/installation.md
+++ b/docs/docs/getting-started/installation.md
@@ -1,4 +1,4 @@
-$---
+---
 id: installation
 title: Installation
 sidebar_label: Installation

--- a/docs/docs/text/fonts.md
+++ b/docs/docs/text/fonts.md
@@ -24,29 +24,3 @@ export const HelloWorld = () => {
   );
 };
 ```
-
-Alternatively, you can use your own set of custom fonts to be available in the canvas, as seen below.
-
-```tsx twoslash
-import {Canvas, Text, useFontMgr} from "@shopify/react-native-skia";
-
-export const HelloWorld = () => {
-  const fontMgr = useFontMgr([
-    require("./my-custom-font.otf")
-  ]);
-  if (fontMgr === null) {
-    return null;
-  }
-  return (
-    <Canvas style={{ flex: 1 }} fontMgr={fontMgr}>
-      <Text
-        x={0}
-        y={0}
-        value="Hello World"
-        familyName="My Custom Font"
-        size={16}
-      />
-    </Canvas>
-  );
-};
-```

--- a/docs/docs/text/text.md
+++ b/docs/docs/text/text.md
@@ -14,6 +14,7 @@ The fonts available in the canvas are described in [here](/docs/text/fonts).
 | value      | `string`  | Text to draw                                                  |
 | familyName | `string`  | Font family name                                              |
 | size       | `number`  | Font size                                                     |
+| font       | `font`    | Custom font to use (loaded with useFont)                      |
 
 ### Example
 
@@ -29,6 +30,31 @@ export const HelloWorld = () => {
         value="Hello World"
         familyName="serif"
         size={32}
+      />
+    </Canvas>
+  );
+};
+```
+
+## Using a custom font
+
+Alternatively, you can use your own set of custom fonts to be available in the canvas, as seen below.
+
+```tsx twoslash
+import {Canvas, Text, useFont} from "@shopify/react-native-skia";
+
+export const HelloWorld = () => {
+  const font = useFont(require("./my-custom-font.otf"), 16);
+  if (font === null) {
+    return null;
+  }
+  return (
+    <Canvas style={{ flex: 1 }}>
+      <Text
+        x={0}
+        y={0}
+        font={font}
+        value="Hello World"
       />
     </Canvas>
   );

--- a/example/src/Examples/Matrix/Matrix.tsx
+++ b/example/src/Examples/Matrix/Matrix.tsx
@@ -3,12 +3,13 @@ import {
   Canvas,
   Fill,
   Paint,
+  useFont,
   useFontMgr,
 } from "@shopify/react-native-skia";
 import React from "react";
 import { useTimestamp } from "@shopify/react-native-skia/src/animation/Animation/hooks";
 
-import { COLS, ROWS, Symbol } from "./Symbol";
+import { COLS, ROWS, Symbol, SYMBOL } from "./Symbol";
 
 const cols = new Array(COLS).fill(0).map((_, i) => i);
 const rows = new Array(ROWS).fill(0).map((_, i) => i);
@@ -32,13 +33,12 @@ const streams = cols.map(() =>
 
 export const Matrix = () => {
   const timestamp = useTimestamp();
-  const fontMgr = useFontMgr([require("./matrix-code-nfi.otf")]);
-  if (fontMgr === null) {
+  const font = useFont(require("./matrix-code-nfi.otf"), SYMBOL.height);
+  if (font === null) {
     return null;
   }
-
   return (
-    <Canvas style={{ flex: 1 }} fontMgr={fontMgr}>
+    <Canvas style={{ flex: 1 }}>
       <Fill color="black" />
       <Paint>
         <BlurMask sigma={10} style="solid" />
@@ -46,6 +46,7 @@ export const Matrix = () => {
       {cols.map((_i, i) =>
         rows.map((_j, j) => (
           <Symbol
+            font={font}
             timestamp={timestamp}
             key={`${i}-${j}`}
             i={i}

--- a/example/src/Examples/Matrix/Symbol.tsx
+++ b/example/src/Examples/Matrix/Symbol.tsx
@@ -1,5 +1,5 @@
 import React, { useRef } from "react";
-import type { AnimationValue } from "@shopify/react-native-skia";
+import type { AnimationValue, Font } from "@shopify/react-native-skia";
 import { Text } from "@shopify/react-native-skia";
 import { Dimensions } from "react-native";
 
@@ -16,9 +16,10 @@ interface SymbolProps {
   j: number;
   timestamp: AnimationValue<number>;
   stream: number[];
+  font: Font;
 }
 
-export const Symbol = ({ i, j, timestamp, stream }: SymbolProps) => {
+export const Symbol = ({ i, j, timestamp, stream, font }: SymbolProps) => {
   const offset = useRef(Math.round(Math.random() * (symbols.length - 1)));
   const range = useRef(100 + Math.random() * 900);
   const x = i * SYMBOL.width;
@@ -35,8 +36,7 @@ export const Symbol = ({ i, j, timestamp, stream }: SymbolProps) => {
     <Text
       x={x + SYMBOL.width / 4}
       y={y + SYMBOL.height}
-      familyName="Matrix Code NFI"
-      size={SYMBOL.height}
+      font={font}
       value={value}
       opacity={opacity}
       color={() =>

--- a/package/cpp/api/JsiSkFontMgrFactory.h
+++ b/package/cpp/api/JsiSkFontMgrFactory.h
@@ -5,9 +5,6 @@
 #include "JsiSkData.h"
 #include <SkFontMgr.h>
 
-// TODO: include proper header instead, this might require a Skia version upgrade?
-sk_sp<SkFontMgr> SkFontMgr_New_Custom_Data(sk_sp<SkData>* datas, int n);
-
 #include <jsi/jsi.h>
 
 namespace RNSkia {
@@ -16,18 +13,18 @@ namespace RNSkia {
 
     class JsiSkFontMgrFactory : public JsiSkHostObject {
     public:
-        JSI_HOST_FUNCTION(FromData) {
-            std::vector<sk_sp<SkData>> data;
-            for (int i = 0; i < count; i++) {
-                auto d = JsiSkData::fromValue(
-                        runtime, arguments[i]);
-                data.push_back(d);
-            }
-
-            auto fontMgr = SkFontMgr_New_Custom_Data(data.data(), data.size());
-            return jsi::Object::createFromHostObject(
-            runtime, std::make_shared<JsiSkFontMgr>(getContext(), fontMgr));
-        }
+//        JSI_HOST_FUNCTION(FromData) {
+//            std::vector<sk_sp<SkData>> data;
+//            for (int i = 0; i < count; i++) {
+//                auto d = JsiSkData::fromValue(
+//                        runtime, arguments[i]);
+//                data.push_back(d);
+//            }
+//
+//            auto fontMgr = SkFontMgr_New_Custom_Data(data.data(), data.size());
+//            return jsi::Object::createFromHostObject(
+//            runtime, std::make_shared<JsiSkFontMgr>(getContext(), fontMgr));
+//        }
 
         JSI_HOST_FUNCTION(RefDefault) {
             auto fontMgr = SkFontMgr::RefDefault();
@@ -36,7 +33,7 @@ namespace RNSkia {
         }
 
         JSI_EXPORT_FUNCTIONS(
-                JSI_EXPORT_FUNC(JsiSkFontMgrFactory, FromData),
+                //JSI_EXPORT_FUNC(JsiSkFontMgrFactory, FromData),
                 JSI_EXPORT_FUNC(JsiSkFontMgrFactory, RefDefault)
         )
 

--- a/package/src/renderer/components/text/Text.tsx
+++ b/package/src/renderer/components/text/Text.tsx
@@ -2,27 +2,40 @@ import React from "react";
 
 import type { CustomPaintProps } from "../../processors/Paint";
 import { useDrawing } from "../../nodes/Drawing";
-import type { AnimatedProps } from "..";
+import type { Font } from "../../../skia";
 import { Skia } from "../../../skia";
+import type { AnimatedProps } from "..";
 
-interface TextProps extends CustomPaintProps {
-  value: string;
-  familyName: string;
-  size: number;
-  x: number;
-  y: number;
-}
+type FontDef = { font: Font } | { familyName: string; size: number };
+
+const isFont = (fontDef: FontDef): fontDef is { font: Font } =>
+  // We use any here for safety (JSI instances don't have hasProperty working properly);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (fontDef as any).font !== undefined;
+
+type TextProps = CustomPaintProps &
+  FontDef & {
+    value: string;
+    x: number;
+    y: number;
+  };
 
 export const Text = (props: AnimatedProps<TextProps>) => {
   const onDraw = useDrawing(
     props,
-    ({ canvas, paint, fontMgr }, { value, x, y, familyName, size }) => {
-      const typeface = fontMgr.matchFamilyStyle(familyName);
-      if (typeface === null) {
-        throw new Error(`No typeface found for ${familyName}`);
+    ({ canvas, paint, fontMgr }, { value, x, y, ...fontDef }) => {
+      let selectedFont: Font;
+      if (isFont(fontDef)) {
+        selectedFont = fontDef.font;
+      } else {
+        const { familyName, size } = fontDef;
+        const typeface = fontMgr.matchFamilyStyle(familyName);
+        if (typeface === null) {
+          throw new Error(`No typeface found for ${familyName}`);
+        }
+        selectedFont = Skia.Font(typeface, size);
       }
-      const font = Skia.Font(typeface, size);
-      canvas.drawText(value, x, y, paint, font);
+      canvas.drawText(value, x, y, paint, selectedFont);
     }
   );
   return <skDrawing onDraw={onDraw} {...props} />;

--- a/package/src/skia/Font/useFont.ts
+++ b/package/src/skia/Font/useFont.ts
@@ -1,22 +1,21 @@
 /*global SkiaApi*/
-import type { DependencyList } from "react";
 import { useMemo } from "react";
 
+import type { DataSource } from "../Data";
 import { Skia } from "../Skia";
-import type { Typeface } from "../Typeface";
+import { useTypeface } from "../Typeface";
 
 import type { Font } from "./Font";
 
 /**
  * Returns a Skia Font object
  * */
-export const useFont = (
-  typeface?: Typeface,
-  size?: number,
-  deps: DependencyList = []
-): Font =>
-  useMemo(() => {
-    if (typeface && size) {
+export const useFont = (font: DataSource, size?: number): Font | null => {
+  const typeface = useTypeface(font);
+  return useMemo(() => {
+    if (typeface === null) {
+      return null;
+    } else if (typeface && size) {
       return Skia.Font(typeface, size);
     } else if (typeface && !size) {
       return Skia.Font(typeface);
@@ -24,4 +23,5 @@ export const useFont = (
       return Skia.Font();
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, deps);
+  }, [typeface]);
+};

--- a/package/src/skia/FontMgr/FontMgrFactory.ts
+++ b/package/src/skia/FontMgr/FontMgrFactory.ts
@@ -1,17 +1,17 @@
-import type { DependencyList } from "react";
+// import type { DependencyList } from "react";
 
-import type { Data, DataSource } from "../Data";
-import { useDataCollection } from "../Data/Data";
-import { Skia } from "../Skia";
+// import type { Data, DataSource } from "../Data";
+// import { useDataCollection } from "../Data/Data";
+// import { Skia } from "../Skia";
 
 import type { FontMgr } from "./FontMgr";
 
 export interface FontMgrFactory {
-  FromData: (...data: Data[]) => FontMgr | null;
+  // FromData: (...data: Data[]) => FontMgr | null;
   RefDefault: () => FontMgr;
 }
 
-const fontMgrFactory = (data: Data[]) => Skia.FontMgr.FromData(...data);
+//const fontMgrFactory = (data: Data[]) => Skia.FontMgr.FromData(...data);
 
-export const useFontMgr = (data: DataSource[], deps: DependencyList = []) =>
-  useDataCollection(data, fontMgrFactory, deps);
+// export const useFontMgr = (data: DataSource[], deps: DependencyList = []) =>
+//   useDataCollection(data, fontMgrFactory, deps);

